tests/native/test_models.py

```diff
@@ -208,6 +203,51 @@ def test_symbolic(self):
         ret = strlen(self.state, s)
         self.assertTrue(self.state.must_be_true(ret == 3))
 
+    def test_symbolic_fork(self):
+        # This binary is compiled using gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
+        # with flags: -g -static -fno-builtin
+        BIN_PATH = os.path.join(os.path.dirname(__file__), "binaries", "sym_strlen_test")
+        tmp_dir = tempfile.TemporaryDirectory(prefix="mcore_test_sym_")
+        m = Manticore(BIN_PATH, stdin_size=10, workspace_url=str(tmp_dir.name))
+
+        addr_of_strlen = 0x04404D0
+
+        @m.hook(addr_of_strlen)
+        def strlen_model(state):
+            state.invoke_model(strlen)
+
+        m.run()
+        m.finalize()
+
+        # Expected stdout outputs
+        expected = [
+            "Length of string is: 0",
+            "Length of string is: 1",
+            "Length of string is: 2",
+            "Length of string is: 3",
+            "Length of string is: 4",
+            "Length of string is: 5",
+        ]
+
+        # Make a list of the generated output states
+        outputs = f"{str(m.workspace)}/test_*.stdout"
+        stdouts = []
+        for out in glob(outputs):
+            with open(out) as f:
+                stdouts.append(f.read())
+
+        # Check the number of output states is the number of expected stdouts
+        self.assertEqual(len(stdouts), len(expected))
+
+        # Assert that every expected stdout has a matching output
+        for e in expected:
+            match = False
+            for s in stdouts:
+                if e == s:
+                    match = True
+                    break
+            self.assertTrue(match)
```

Comment:
Nit: Since we're comparing simple strings this time (as opposed to matching regex in the `strcpy` test), we can use sets to easily and efficiently compare `expected` vs. `output`

Reply:
Updated :)